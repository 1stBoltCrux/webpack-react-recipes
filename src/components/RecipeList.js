import React from 'react';
import PropTypes from 'prop-types';


const RecipeList = (props) => {
  const { recipes } = props;
  return (
    <div>
      <h2>Recipes</h2>
      <ul className="list-reset">
        {recipes.map(recipe => (

          <li className="py2 border-bottom border-bottom-dashed pointer" key={recipe.id} onClick={() => props.onClick(recipe.id)}>
            <span>{recipe.name}</span>
            <span>{recipe.category}</span>
          </li>

        ))}
      </ul>
    </div>
  );
};


RecipeList.propTypes = {
  recipes: PropTypes.object,
  // styles: PropTypes.string,
  onClick: PropTypes.func,
};

export default RecipeList;
