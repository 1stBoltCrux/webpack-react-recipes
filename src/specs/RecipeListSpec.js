import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';


const testRecipes = [
  {
    id: 1,
    name: 'recipe 1',
    category: 'category 1'
  }
]

describe('<RecipeList />', () => {
  test('Should not break when no recipes are passed', () => {
    const component = renderer.create(

        <RecipeList/>

    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
  test('Should render correctly', () => {
    const component = renderer.create(

        <RecipeList recipes={testRecipes}/>

    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
  test('Should render favorited state correctly', () => {
    const component = renderer.create(

        <RecipeList recipes={testRecipes} favorited={true}/>

    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
})
