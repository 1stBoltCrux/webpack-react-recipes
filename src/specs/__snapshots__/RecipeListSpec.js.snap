// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should not break when no recipes are passed 1`] = `
<div>
  <ul
    className="list-reset"
  />
</div>
`;

exports[`<RecipeList /> Should render correctly 1`] = `
<div>
  <ul
    className="list-reset"
  >
    <li
      className=" py2 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ♡
      </span>
      <span>
        recipe 1
      </span>
      <span>
        category 1
      </span>
    </li>
  </ul>
</div>
`;

exports[`<RecipeList /> Should render favorited state correctly 1`] = `
<div>
  <ul
    className="list-reset"
  >
    <li
      className=" py2 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ❤️
      </span>
      <span>
        recipe 1
      </span>
      <span>
        category 1
      </span>
    </li>
  </ul>
</div>
`;
